react-components/hello-world.md

```diff
@@ -0,0 +1,386 @@
+
+1. `src/index.js`
+1. `src/App.js`
+
+### Exploring `index.js`
+
+Let's go ahead and open this project up in our text editor to examine the files that have been created for us. We'll begin with `src/index.js` and practice our reading skills.
+
+```js
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import reportWebVitals from "./reportWebVitals";
+
+ReactDOM.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById("root")
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
+```
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: checkbox
+* id: e447ec71
+* title: Hello, World!
+##### !question
+
+`src/index.js` imports several files. Select the files that are imported.
+
+##### !end-question
+##### !options
+
+* `React`
+* `ReactDOM`
+* `'./index.css'`
+* `App` from `'./App'`
+* `logo` from `'./logo.svg'`
+* `'./index.js'`
+
+##### !end-options
+##### !answer
+
+* `React`
+* `ReactDOM`
+* `'./index.css'`
+* `App` from `'./App'`
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: 855add45
+* title: Hello, World!
+##### !question
+
+`ReactDOM.render( ... );` is a...
+
+##### !end-question
+##### !options
+
+* variable
+* function call
+* class
+
+##### !end-options
+##### !answer
+
+* function call
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 4 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: b56db597
+* title: Hello, World!
+##### !question
+
+The following code snippet is in `src/index.js`.
+
+```js
+<React.StrictMode>
+  <App />
+</React.StrictMode>
+```
+
+This code is...
+
+##### !end-question
+##### !options
+
+* An import statement
+* A comment in React
+* The first argument when calling `ReactDOM.render()`
+* An object literal
+
+##### !end-options
+##### !answer
+
+* The first argument when calling `ReactDOM.render()`
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+`index.js` is responsible for being the starting point of our webapp.
+
+It is the entry point of our project; our app is configured to look for this file first when we run the app.
+
+This file renders our first _JSX_ (described in another lesson).
+
+```js
+<React.StrictMode>
+  <App />
+</React.StrictMode>
+```
+
+This code indicates that when we start our React webapp, it will _render_ an `App` component using `<App />`.
+
+In this curriculum, we will rarely touch `src/index.js`. Additionally, there are other lines of code in this file that is worth exploring with bigger projects.
+
+Instead, we'll focus on leveraging the `App` component.
```

Comment:
acknowledge this rare current moment of visiting the file and what we got out of it. demote the part about the bigger projects to a sub-clause so that the "Instead" can more easily tie back to the idea of rarely coming back to this file. "leveraging" sounds odd since that usually means to make use of something that exists, whereas we'll need to do a lot of work to build up the App component.
```suggestion
Aside from seeing that this is how the `App` component gets loaded, we will rarely visit `src/index.js` in this curriculum, even though there are lines of code in this file that are worth exploring with bigger projects.

Instead, we'll focus on working with the `App` component.
```